### PR TITLE
Set includeShardsStats = false in NodesStatsRequest where the caller does not use shards-level statistics

### PR DESCRIPTION
**Pull request by NEUpanning**
Relates #100466

### CODE REVIEW

File: docs/changelog/100938.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 100938
+summary: "Set includeShardsStats = true in NodesStatsRequest where the caller does not use shards-level statistics"
```
Comment (gmarouli) [inline docs/changelog/100938.yaml:2]: Just to double check, do you mean here `includeShardsStats = false`?

**Comment by gmarouli** [comment]
@elasticmachine ok to test

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Comment by gmarouli** [comment]
buildkite test this please

**Comment by gmarouli** [comment]
@NEUpanning thank you for leading this! This is such a great improvement. The code looks great. I just triggered the tests now, I see no blockers on moving forward with this. Let's see if the tests agree :)

**Comment by gmarouli** [comment]
@NEUpanning if you need any help addressing the format failures, let me know

File: docs/changelog/100938.yaml (hunk 1 above)
Comment (NEUpanning) [reply docs/changelog/100938.yaml]: Yes, it's my mistake. Thanks for your check. It's fixed in f72732ea279284ddc62712184ae139fa3a3c6e60 and this PR title is also fixed.

**Comment by NEUpanning** [comment]
I resolved  the format failures. @gmarouli Could you trigger the tests please?

**Comment by gmarouli** [comment]
buildkite test this please

**Review by gmarouli — APPROVED** [review]
LGTM 🚀 , thank you @NEUpanning for your contribution!

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Comment by gmarouli** [comment]
buildkite test this please

**Comment by NEUpanning** [comment]
@gmarouli Thank you for reviewing my code.

**Pull request merged**